src/java.base/share/classes/java/util/zip/GZIPInputStream.java

```diff
@@ -95,8 +95,16 @@ public GZIPInputStream(InputStream in) throws IOException {
      * Reads uncompressed data into an array of bytes. If {@code len} is not
      * zero, the method will block until some input can be decompressed; otherwise,
      * no bytes are read and {@code 0} is returned.
+     * <p>
+     * If <i>n</i> denotes the returned number of inflated bytes then {@code buf[off]}
+     * through {@code buf[off+}<i>n</i>{@code -1]} will contain the uncompressed data. The
+     * elements {@code buf[off+}<i>n</i>{@code ]} through {@code buf[off+}<i>len</i>{@code -1]}
+     * are undefined and an implementation is free to change them during the inflate
+     * operation. If the return value is -1 or an exception is thrown the whole
+     * content of {@code buf} is undefined.
+     *
      * @param buf the buffer into which the data is read
-     * @param off the start offset in the destination array {@code b}
+     * @param off the start offset in the destination array {@code buf}
      * @param len the maximum number of bytes read
      * @return  the actual number of bytes read, or -1 if the end of the
      *          compressed input stream is reached
```

Comment:
We should probably adjust the `@return` description, as was done for InflaterInputStream, otherwise the method. description will say "the returning number of inflated bytes" and the return description will say "the actual number of bytes read".

Reply:
Fixed (also in `ZipInputStream`).